doc/iterations/2024.11/add_support_for_form_property_section_layout.adoc

```diff
@@ -0,0 +1,66 @@
+= Add support for form property section layout
+
+== Problem
+
+A property section in a form is made of a label, a help icon and a value area.
+Currently, the layout is always vertical: label and help icon are in div and the value area is below.
+
+== Key Result
+
+It will be possible as a specifier to choose if the layout is horizontal or vertical.
+
+== Solution
+
+The property section contains 
+
+* a "label" div which includes the label and the help icon.
+* a "value"div
+
+The content of the property section is layouted with CSS flexbox.
+New attributes are added on form widget styles to specify the layout.
+It corresponds to CSS flexbox options:
+
+ * flexDirection
+ * labelFlex : the flex options which is a shortHand for for `flex-grow `, `flex-shrink` and  `flex-basis`
+ * valueFlex : the flex options which is a shortHand for for `flex-grow `, `flex-shrink` and  `flex-basis`
+ * gap
+
+Example:
+If many property sections are defined with `labelFlex='0 0 150px'`, and 'flexDirection=row', the "label" div of all the property sections will be justified.
```

Comment:
I really don't understand. The expect key result is to choose a horizontal or vertical layout. Now you seem to be want to expose obscure flexbox options (flex-basis, flex-shrink) which are not relevant to this goal to provide a poor way to control the layout of the second dimension.

If you want a horizontal layout with `flexDirection=row` AND add some constraints on the vertical side too, use grid. Otherwise don't add poor capabilities to control the other dimension.

Either you need to control both dimensions and in this case provide use the dedicated API (grid) or you don't and then don't provide it. You need to clarify that from the key result part directly.

Reply:
Effectively, I need both direction.
So I started using CSS grid.
The problem is I did not find elegant solution to have the equivalent of Flexbox `row-reverse` and `column-reverse` and I need it to be iso-functional with Checkbox. (Otherwise I would not care) which already have labelPlacement attribute.

The aim is having the upper part of the following screenshot
![image](https://github.com/user-attachments/assets/268c75ea-d5cc-4f7e-bde3-858a9e9f91b4)